### ISSUE TITLE
Multiple user logins running multiple test

### ISSUE THREAD

**Issue by jarrettj** [Codeception/Codeception#4564]
#### What are you trying to achieve?
Busy testing user creation. If I run the test alone as a single test it works. If I use the -g option running with other tests it fails. If I run the folder with other tests it fails.

#### What do you get instead?
Should have the same outcome whether it's a single test or multiple tests.

> Provide console output if related. Use `-vvv` mode for more details.

```bash
php vendor/bin/codecept run --env docker-local-chrome-2 -g paracept_2 -vvv
[Groups] paracept_2 

  [Db] Connected to ci_test
  [Db] Connected to ci_test
Acceptance (docker-local-chrome-2) Tests (5) ------------------------------------------------------------------------------------------------------------------------------------
Modules: WebDriver, Db, Redis
---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
CreateCept: Create an admin user.
Signature: CreateCept
Test: module/Application/tests/acceptance/Admin/User/CreateCept.php
Scenario --
 I login "admin-user@mailinator.com","password"
   I load session snapshot "admin-user@mailinator.com"
   I am on page "/"
  [GET] http://webserver.p2.docker/
   I fill field "identity","admin-user@mailinator.com"
   I fill field "credential","password"
   I submit form ".form-signin",[]
  [Uri] http://webserver.p2.docker/user/login?redirect=/
  [Method] post
  [Parameters] []
  [Page] /index/litigation
   I save session snapshot "admin-user@mailinator.com"
  [Snapshot] Saved "admin-user@mailinator.com" session snapshot
 I am on page "admin-user/add"
  [GET] http://webserver.p2.docker/admin-user/add
 I can see in current url "admin-user/add"
 I can see "Add a new User"
 I fill field "input[name="username"]","Test User new"
 I fill field "input[name="displayName"]","Test User new"
 I fill field "input[name="email"]","test-user-new@mailinator.com"
 I fill field "input[name="password"]","password"
 I select option "select[name="company"]","Default Company"
 I execute js "$('#User > div:nth-child(8) > div > div > div:nth-child(2) > label > div > input').iCheck('check');"
 I submit form "form#User",[]
  [Uri] http://webserver.p2.docker/admin-user/add
  [Method] post
  [Parameters] []
  [Page] /admin-user/edit/2
 I wait for element "form#User",30
 I can see "Edit User"
 I can see in field "input[name="username"]","Test User new"
 I can see in field "input[name="displayName"]","Test User new"
 I can see in field "input[name="email"]","test-user-new@mailinator.com"
 I can see option is selected "select[name="company"]","Default Company"
 I am on page "admin-user"
  [GET] http://webserver.p2.docker/admin-user
 I wait for element visible "select[name="data-table_length"]",30
 I select option "select[name="data-table_length"]","All"
 I can see "Test User new"
 I can see "test-user-new@mailinator.com"
 I can see "Default Company"
 PASSED 

CreateLoginCept: Login as a new admin user.
Signature: CreateLoginCept
Test: module/Application/tests/acceptance/Tasks/User/CreateLoginCept.php
Scenario --
 I login "admin-user@mailinator.com","password"
   I load session snapshot "admin-user@mailinator.com"
  [Snapshot] Restored "admin-user@mailinator.com" session snapshot
 I am on page "admin-user/add"
  [GET] http://webserver.p2.docker/admin-user/add
 I can see in current url "admin-user/add"
 I can see "Add a new User"
 I fill field "input[name="username"]","Test User admin login"
 I fill field "input[name="displayName"]","Test User admin login"
 I fill field "input[name="email"]","test-user-admin-login@mailinator.com"
 I fill field "input[name="password"]","testnewpassword"
 I select option "select[name="company"]","Default Company"
 I execute js "$('#User > div:nth-child(8) > div > div > div:nth-child(2) > label > div > input').iCheck('check');"
 I submit form "form#User",[]
  [Uri] http://webserver.p2.docker/admin-user/add
  [Method] post
  [Parameters] []
  [Page] /admin-user/edit/3
 I wait for element "form#User",30
 I can see "Edit User"
 I can see in field "input[name="username"]","Test User admin login"
 I can see in field "input[name="displayName"]","Test User admin login"
 I can see in field "input[name="email"]","test-user-admin-login@mailinator.com"
 I can see option is selected "select[name="company"]","Default Company"
 I am on page "admin-user"
  [GET] http://webserver.p2.docker/admin-user
 I wait for element visible "select[name="data-table_length"]",30
 I select option "select[name="data-table_length"]","All"
 I can see "Test User admin login"
 I can see "test-user-admin-login@mailinator.com"
 I can see "Default Company"
 I logout 
   I click ".logout"
   I wait for element ".login-form",30
 I login "test-user-admin-login@mailinator.com","testnewpassword"
   I load session snapshot "test-user-admin-login@mailinator.com"
   I am on page "/"
  [GET] http://webserver.p2.docker/
   I fill field "identity","test-user-admin-login@mailinator.com"
   I fill field "credential","testnewpassword"
   I submit form ".form-signin",[]
  [Uri] http://webserver.p2.docker/user/login?redirect=/
  [Method] post
  [Parameters] []
  [Page] /user/login?redirect=/
   I save session snapshot "test-user-admin-login@mailinator.com"
  [Snapshot] Saved "test-user-admin-login@mailinator.com" session snapshot

Time: 30.45 seconds, Memory: 16.00MB

There were 4 failures:

---------
1) CreateLoginCept: Login as a new admin user.
 Test  module/Application/tests/acceptance/Tasks/User/CreateLoginCept.php
 Step  Can see "XXXXX"
 Fail  Failed asserting that  on page /user/login?redirect=/
-->
Sign in
Forgot password?
--> contains "XXXXX".

```
> Provide test source code if related

```php
// @group admin
// @group user
// @group login
use Tests\Constants;

$I = new WebGuy($scenario);
$I->login(Constants::USERNAME, Constants::PASSWORD);
$I->wantTo('Login as a new admin user.');

# Create admin user
$userName = 'Test User admin login';
$userEmail = 'test-user-admin-login@mailinator.com';
$userType = 'admin';
$userCompany = 'Default Company';
$userRole = 'admin';
$userPassword = 'testnewpassword';
        
require realpath(\Codeception\Util\Fixtures::get('accDir') . '/Admin/User/helpers/Create.php');

$I->logout();
$I->login($userEmail, $userPassword);
$I->canSee('XXXXX');

  /**
     * Define custom actions here
     */
    public function login($username, $password)
    {
        $I = $this;
        // if snapshot exists - skipping login
        if ($I->loadSessionSnapshot($username)) {
            return;
        }
        // logging in
        $I->amOnPage('/');
        $I->submitForm('.form-signin', [
            'identity' => $name,
            'credential' => $password
        ]);
         // saving snapshot
        $I->saveSessionSnapshot($username);
    }
```
### Details

* Codeception version: 2.3.6
* PHP Version: 7.0.22
* Operating System: Ubuntu
* Installation type: Composer
* List of installed packages (`composer show`)
* Suite configuration:

```yml
codeception.yml
paths:
    tests: module/Application/tests
    log: module/Application/tests/_log
    data: module/Application/tests/_data
    helpers: module/Application/tests/_helpers
settings:
    bootstrap: _bootstrap.php
    suite_class: \PHPUnit_Framework_TestSuite
    colors: true
    memory_limit: 1024M
    log: true
modules:
    config:
        Db:
            dsn: 'mysql:host=localhost;dbname=ci_test'
            user: 'root'
            password: 'root'
            dump: module/Application/tests/_data/dump.sql
coverage:
    enable: true
    remote: true
    include:
        - module/Application/src/*
groups:
    paracept_*: module/Application/tests/_data/robo/paracept_*

acceptance.yml
class_name: WebGuy
modules:
    enabled:
        - WebDriver
        - Db
        - Redis
env:
    docker-local-chrome-2:
        modules:
            config:
                WebDriver:
                    url: 'http://webserver.p1.docker'
                    browser: chrome
                    host: p2_selenium-hub_1
                    clear_cookies: false
                    restart: false
                Db:
                    dsn: 'mysql:host=p2_mysql_1;dbname=ci_test'
                    user: 'root'
                    password: 'root'
                    dump: 'module/Application/tests/_data/dump.sql'
                    populate: true
                    cleanup: false
                Redis:
                    host: 'p1_redis_1'
                    port: 6379
                    database: 0
```

It's like it can't login after login out. Why would session just vanish? 

Regards.
JJ

**Comment by Naktibalda** [comment]
Probably caused by `// if snapshot exists - skipping login`. After loading sessoon snapshot no page is loaded.

**Comment by jarrettj** [comment]
I tried removing the session snapshot bit. Still fails with the same error. As soon as I logout, I can't log back in. Will do some more digging around. Maybe add some logging in the application. Thanks

**Comment by jarrettj** [comment]
No luck at all. Thought I would give it another go. It works if I run the test alone. Atleast I got that far. Just a single run. When running multiple tests it fails though. Well, if it is not the first test it fails.

If I access the site manually and login, create a user, logout, login as new user it works fine. For some obscure reason, I cannot log back in after I log out in the test suite. Tried everything.
